TwoPerformant/Feed/Model/Cron.php

```diff
@@ -0,0 +1,91 @@
+<?php
+namespace TwoPerformant\Feed\Model;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\FilterBuilder;
+use Magento\Store\Model\StoreManagerInterface;
+
+
+class Cron
+{
+    protected $productRepository;
+    protected $searchCriteriaBuilder;
+    protected $filterBuilder;
+    protected $storeManager;
+
+    public function __construct(
+        ProductRepositoryInterface $productRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        FilterBuilder $filterBuilder,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->productRepository = $productRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->filterBuilder = $filterBuilder;
+        $this->storeManager = $storeManager;
+    }
+
+    public function export()
+    {
+        $date = $this->getLastWeekDate();
+        $items = $this->getProducts($date);
+        $this->writeToFile($items);
+    }
+
+    protected function getLastWeekDate()
+    {
+        $now = new \DateTime();
+        $interval = new \DateInterval('P1W');
+        $lastWeek = $now->sub($interval);
+        return $lastWeek;
+    }
+
+    public function getProducts($date)
+    {
+        $filters = [];
+
+        $filters[] = $this->filterBuilder
+            ->setField('created_at')
+            ->setConditionType('gt')
+            ->setValue($date->format('Y-m-d H:i:s'))
+            ->create();
+
+        $this->searchCriteriaBuilder->addFilters($filters);
+
+        $searchCriteria = $this->searchCriteriaBuilder->create();
+        $searchResults = $this->productRepository->getList($searchCriteria);
+        return $searchResults->getItems();
+    }
+
+    protected function writeToFile($items)
+    {
+        if (count($items) > 0) {
+            $feed = fopen("pub/feeds/2p_feed.csv", "w") or die("Unable to open file!");
+            $mediaUrl = $this->storeManager->getStore()->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_MEDIA);
+
+            foreach ($items as $item) {
+                $category = $item->getCategory() ? $item->getCategory() : 'default';
+
+                if($item->getPrice()) {
```

Comment:
in what context(s) would a product not have a price?

Reply:
In magento you can set a different prices for the same product (and SKU) but different sizes/colors. For that case, the price property is null, and the prices must be computed in a different way. I couldn't figure it out exactly how or if we can use that information somehow. In the UI, the prices are shown as "starting from ..."

Reply:
https://webkul.com/blog/get-price-range-configurable-product-magento-2/ ?